cpu/native/include/cpu.h

```diff
@@ -34,14 +35,14 @@ extern "C" {
 #define CPU_HAS_UNALIGNED_ACCESS
 
 /**
- * @brief   Prints the address the callee will return to
+ * @brief   Gets the address the callee will return to
  */
-__attribute__((always_inline)) static inline void cpu_print_last_instruction(void)
+__attribute__((always_inline)) static inline uintptr_t cpu_get_caller_pc(void)
 {
     /* __builtin_return_address will return the address the calling function
-     * will return to - since cpu_print_last_instruction is forced inline,
+     * will return to - since cpu_get_caller_pc is forced inline,
      * it is the return address of the user of this function */
-    printf("%p\n", __builtin_return_address(0));
+    return (uintptr_t)__builtin_return_address(0);
```

Comment:
Wow, does this exist for other arches as well? This would be the most convenient implementation there is :)

But this could be a follow up.

Reply:
https://gcc.gnu.org/onlinedocs/gcc/Return-Address.html

Reply:
Just from the return value type it ia obvious this won't work on AVR.

And the warning about this builtin potentially crashing the program is not super comforting. But I think on Xtensa and ARM this should work fine for level 0.

Reply:
it think non 0 crashing would also happen to other implementations (but if there is a bunch of assembly you are not warned about this) level 0 should be safe for any machine (may still not return a usable value but not crash)